python/mxnet/visualization.py

```diff
@@ -134,17 +134,23 @@ def print_layer_summary(node, out_shape):
                             pre_filter = pre_filter + int(shape[0])
         cur_param = 0
         if op == 'Convolution':
-            if ("no_bias" in node["attrs"]) and int(node["attrs"]["no_bias"]):
-                cur_param = pre_filter * int(node["attrs"]["num_filter"])
+            if ("no_bias" in node["attrs"]) and node["attrs"]["no_bias"] == 'True':
+                num_group = int(node["attrs"]["num_group"]) if \
+                   ("num_group" in node["attrs"]) else 1
+                cur_param = (pre_filter * int(node["attrs"]["num_filter"])) \
+                   // num_group
                 for k in _str2tuple(node["attrs"]["kernel"]):
                     cur_param *= int(k)
             else:
-                cur_param = pre_filter * int(node["attrs"]["num_filter"])
+                num_group = int(node["attrs"]["num_group"]) if \
+                   ("num_group" in node["attrs"]) else 1
+                cur_param = (pre_filter * int(node["attrs"]["num_filter"])) \
+                   // num_group
                 for k in _str2tuple(node["attrs"]["kernel"]):
                     cur_param *= int(k)
                 cur_param += int(node["attrs"]["num_filter"])
         elif op == 'FullyConnected':
-            if ("no_bias" in node["attrs"]) and int(node["attrs"]["no_bias"]):
+            if ("no_bias" in node["attrs"]) and node["attrs"]["no_bias"] == 'True':
```

Comment:
`if bool_str == 'True' ` -> `if bool(bool_str)`

Reply:
Are you sure about this?
bool("False") is True

Reply:
you're right. @chinakook sorry for the misleading comment.
@piiswrong BTW do you happen to know why we are not using json's built-in boolean type?

Reply:
I think i can replace bool(bool_str) with built-in function eval(bool_str).

Reply:
That would create an injection point, since the files may not be trusted. (e.g. `"no_bias": "import shutil; shutil.rmtree('/')"`)

Reply:
I see. Then use 
```python 
node["attrs"]["no_bias"] == 'True' 
``` 
back?

Reply:
yes

Reply:
The json dump should have made booleans without the parenthesis so that the json boolean type is used, i.e.
```
"attrs": {
  "flatten": "True",
  "no_bias": "False",
  "num_hidden": "1000"
}
# should have been
"attrs": {
  "flatten": True,
  "no_bias": False,
  "num_hidden": "1000"
}
```

@piiswrong would it be OK to change json export to properly use json boolean?

Reply:
no. there is compatibility issue